doc/source/dev/data_types.md

```diff
@@ -246,14 +246,34 @@ entries inside your class like this:
 class GenBank(data.Text):
     file_ext = "genbank"
 
-    MetadataElement(name="number_of_sequences", default=0, desc="Number of sequences", readonly=True, visible=True, optional=True, no_value=0)
+    MetadataElement(name="number_of_sequences", default=0, desc="Number of sequences", readonly=True, visible=True, optional=True, no_value="?")
 ```
 
-Here we have a `MetadataElement`, accessible in
-methods with a dataset parameter from
-`dataset.metadata.number_of_sequences`. There are a
-couple relevant functions you'll want to override
-here:
+Here we have a `MetadataElement`, accessible in methods with a dataset parameter
+from `dataset.metadata.number_of_sequences`. 
+
+- `default` (default is `None`): the default value that is set if neither
```

Comment:
Is that right ? Where's that happening ?

Reply:
I guess here https://github.com/galaxyproject/galaxy/blob/6df8d1ae5b54bbb0961576622066a3d2b95f6728/lib/galaxy/model/metadata.py#L134 see also docstring here https://github.com/galaxyproject/galaxy/blob/6df8d1ae5b54bbb0961576622066a3d2b95f6728/lib/galaxy/model/metadata.py#L73